### PR TITLE
Src layout

### LINKED ISSUE #1086: Bootstrap no longer working?
I tried with python 3.10 and 3.8 but no luck. I used to be able to do

```
python -m pip install -r requirements.txt cython
python ./bootstrap.py -m PyMca5.PyMcaGui.pymca.PyMcaMain
```

But now I get

```python
PyMca is installed in /users/denolf/.local/wdnbashutils/virtualenvs/pybox_NTK532/lib/python3.8/site-packages/ 
PyMca data files are installed in PyMca5/PyMcaData 
HTML help files are installed in PyMca5/PyMcaData 
WARNING:bootstrap:`lib` directory does not exist, trying common Python3 lib
Traceback (most recent call last):
  File "./bootstrap.py", line 271, in <module>
    main(sys.argv)
  File "./bootstrap.py", line 231, in main
    runpy.run_module(module, run_name="__main__", alter_sys=True)
  File "/users/denolf/.local/wdnbashutils/.pyenv/versions/3.8.19/lib/python3.8/runpy.py", line 203, in run_module
    mod_name, mod_spec, code = _get_module_details(mod_name)
  File "/users/denolf/.local/wdnbashutils/.pyenv/versions/3.8.19/lib/python3.8/runpy.py", line 111, in _get_module_details
    __import__(pkg_name)
  File "/home/denolf/dev/pymca/PyMca5/__init__.py", line 52, in <module>
    raise ImportError('PyMca cannot be imported from source directory')
ImportError: PyMca cannot be imported from source directory
```

An editable install doesn't work either. How do you currently develop PyMca?

### PR DESCRIPTION
**Pull request by woutdenolf**
Closes #1086

bootstrap.py is not needed anymore so I guess we can remove it as well.

### CODE REVIEW

**Comment by vasole** [comment]
Why are you forcing numpy < 2 in CONTRIBUTING.md?

Is it possible to stick to rst instead of markdown?

**Comment by woutdenolf** [comment]
Because I branched of the master branch from more than a year ago. Rebased and converted md to rest.

**Comment by woutdenolf** [comment]
I forgot the MANIFEST.in file as well.

**Comment by vasole** [comment]
Line 267 in setup.py needs to be corrected.

File: setup.py [hunk 1]
```diff
@@ -168,72 +177,102 @@ def use_gui():
...
+        package_data['PyMca5'] = [os.path.join('PyMcaMath', 'sift', '*.cl')]
 sources = glob.glob('*.c')
+
+script_files = glob_pymca5('scripts', '*')
 if sys.platform == "win32":
     define_macros = [('WIN32', None)]
-    script_files = glob.glob('PyMca5/scripts/*')
-    script_files += glob.glob('scripts/*.bat')
-    script_files.append('scripts/pymca_win_post_install.py')
+    script_files += glob.glob('scripts', '*.bat')
```
Comment (vasole) [inline setup.py:267]: script_files += glob.glob(os.path.join('scripts', '*.bat'))

**Comment by vasole** [comment]
The https://gitlab.esrf.fr/silx/bob/PyMca5 and https://gitlab.esrf.fr/silx/bob/pymca-dmg pipelines will need to be tested too.

**Pull request merged**